src/Symfony/Component/Process/Process.php

```diff
@@ -1632,6 +1634,30 @@ private function escapeArgument(?string $argument): string
         return '"'.str_replace(['"', '^', '%', '!', "\n"], ['""', '"^^"', '"^%"', '"^!"', '!LF!'], $argument).'"';
     }
 
+    private function replacePlaceholders(string $commandline, array $env)
+    {
+        preg_match('/\"\$?([_a-zA-Z0-9]++)?\"/', $commandline, $m);
```

Comment:
why the `?` in the regex? no need for the backslash before double quotes
this replaces only the first match. We should use preg_replace_callback instead
the name should match `[_a-zA-Z]++[_a-zA-Z0-9]*+` (no leading digit)

Reply:
for preg_replace_callback see https://github.com/symfony/symfony/pull/32126#discussion_r296152346